### PR TITLE
Unify the blockquote parsing logic for messages and rawInput

### PR DESCRIPTION
**Pull request by Skalakid**
This PR changes and unifies the blockquote parsing logic for the NewDot messages and the Live Markdown Input. Currently, the parsing result is different when the `shouldKeepRawInput` variable is `true`. Thanks to the changes in this PR, the results are consistent (visually). The only difference is that the tags are being grouped when `shouldKeepRawInput=false`.

Example:
```
> test
>> test
> test
``` 

NewDot sent message `shouldKeepRawInput=false`
```
<blockquote>test<br /><blockquote>test</blockquote>test</blockquote>
```

Live Markdown Input `shouldKeepRawInput=true`
```
<blockquote> test</blockquote>
<blockquote><blockquote> test</blockquote></blockquote>
<blockquote> test</blockquote>
```

Also, it's worth mentioning that new blocquote parsing rules has been discussed [here](https://swmansion.slack.com/archives/C049HHMV9SM/p1731610472261179?thread_ts=1731423572.626789&cid=C049HHMV9SM) and now the following cases are parsed:
```
Valid blocquote examples 
> test
> 
>> test
>>> test
> # heading

Invalid blocquote examples 
> > test last ">" shouldn't be parsed as a blocquote)
> >> test (last ">>" shouldn't be parsed as a blocquote)
>> > test (last ">" shouldn't be parsed as a blocquote)
>>>> test (last ">" shouldn't be parsed as a blocquote)
># heading
```

### Fixed Issues
$ https://github.com/Expensify/App/issues/45154
$ https://github.com/Expensify/App/issues/47951

# Tests
1. Applying changes to both the NewDot app and the Live Markdown Input
2. Verify if the markdown is the same when writing the message and after sending it

# QA
Same as Tests

### CODE REVIEW

**Comment by Pujan92** [comment]
commenting for the assignment.

**Comment by Skalakid** [comment]
Hi @pecanoro, can you assign @Pujan92 to review this PR please?

**Comment by pecanoro** [comment]
Hmm for some reason I can't add him tot he issue even if he commented

**Review by pecanoro — CHANGES_REQUESTED** [review]

File: lib/ExpensiMark.ts [hunk 1]
```diff
@@ -1273,6 +1285,29 @@ export default class ExpensiMark {
         return textToCheck;
     }
 
+    replaceQuoteText(text: string, shouldKeepRawInput: boolean): {replacedText: string; shouldAddSpace: boolean} {
```
[the remark this answers is not in the document]
Comment (Skalakid) [reply lib/ExpensiMark.ts:1222]: Done!

**Comment by Pujan92** [comment]
@Skalakid I am trying to link this commit in E/App but facing some issue while running the app. Wondering how you connected and tested in E/App?

**Comment by Skalakid** [comment]
@Pujan92 expensify-common commits can't be linked to the E/App, but you can add my changes by running `npm i` and then `npm run  build` in the expensify-common repo. After this, copy the JS code from the `/dist/ExpensifyMark.js` file into the `/node_modules/expensify-common/dist/ExpensiMark.js` file in NeeDot

**Comment by Pujan92** [comment]
@Skalakid Isn't the message here should be the same as of live markdown input(Applied a space between 2 blockquotes)?

https://github.com/user-attachments/assets/c7cba1d1-3c3b-4f99-b1f1-14d4a6d474f6

**Comment by Skalakid** [comment]
@Pujan92 Oh, I forgot to tell you one more detail to set up the testing environment, sorry 😅 You must also apply these ExpensiMark changes inside the Live Markdown Library. To do that, you have to:
1. Clone react-native-live-markdown repo
2. Go to the `@Skalakid/fix-blockquote-formatting` branch. I've updated and committed changes for ExpensiMarkd inside Live Markdown parser 
3. Run `yarn install` 
4. `yarn prepare`
5. `yarn pack` - it will generate a packed Live Markdown library file that you can install in the NewDot App
6. Go to the NewDot repo, and in the console, run `npm i -S <path_to_the_previously_generated_tgz_file>`
8. Run `npm i`

And now you will have new parsing logic applied both in the NewDot and Live Markdown libraries :D

**Comment by Skalakid** [comment]
Also, it's worth mentioning here that new rules for the blockquote has been discussed here and now this "> > >" won't be parsed as a nested blocquote

**Comment by Skalakid** [comment]
I updated the PR description with new rules explanation

**Review by pecanoro — APPROVED** [review]

**Comment by pecanoro** [comment]
@Pujan92 I can't assign you for review to this PR but let me know when you have tested it and you can just leave a LGTM and I will merge it!

**Review by Pujan92 — APPROVED** [review]
LGTM, Thanks @Skalakid!

**Pull request merged**